compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -418,10 +418,14 @@ impl<'context> Elaborator<'context> {
 
             if hir_ident.id != DefinitionId::dummy_id() {
                 match self.interner.definition(hir_ident.id).kind {
-                    DefinitionKind::Function(id) => {
+                    DefinitionKind::Function(func_id) => {
                         if let Some(current_item) = self.current_item {
-                            self.interner.add_function_dependency(current_item, id);
+                            self.interner.add_function_dependency(current_item, func_id);
                         }
+
+                        let variable = DependencyId::Variable(hir_ident.location);
```

Comment:
To make things more concrete, the issue is that this location contains the span for the full path rather than just the relevant segment.

Reply:
Thank you for pointing the exact location that needed a change!

I just pushed a commit that fixes this. I also reorganized the tests a bit (I hope it's fine) because I needed to write a new test that was exactly like the one you wrote :-)